### PR TITLE
KOGITO-889: Uniform names for process/rules data definitions (WIP)

### PR DESCRIPTION
**Pull request by evacchi**

### CODE REVIEW

File: api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java [hunk 1]
```diff
@@ -13,15 +13,17 @@
...
 import java.util.Map;
 
+import org.kie.kogito.DataContext;
+
 /**
  * Represents data model type of objects that are usually descriptor of data holders.
  *
  */
-public interface Model {
+public interface ProcessData extends DataContext {
```
Comment (mswiderski) [inline api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: This is backward incompadible change that will break most of already created applications, mostly around tests that rely on Model as generic type

Comment (evacchi) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: I know, this is why I would like to do this before 1.0

Comment (evacchi) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: in fact, I would even go as far as just using DataContext and share it between processes and rule units

Comment (mswiderski) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: so why not leaving it as model? it is generic enough name in my opinion and the Context is way overloaded term especially in our code base

Comment (evacchi) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: that is true, but also `model` is quite overloaded :/

Comment (mswiderski) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessData.java]: don't think it is extensively used within our code base ... overall Model is overloaded but provides bit more info than context

**Comment by evacchi** [comment]
will redo for 0.10+

**Pull request closed without merging**